Runtime/Device/ButtonRing/AdxHIDButtonRing.cs

```diff
@@ -137,6 +140,51 @@ public override void ReadData(ReadOnlySpan<byte> data)
 
             data.CopyTo(_currentState);
         }
+        public unsafe override void ReadDataWithDebounce(IntPtr pointer)
+        {
+            /*
+                if the code below causes any crashes or issues it might be better to 
+                change this function to safe and copy the bytes this way.
+                This is much slower though:
+
+                byte[] currentInput = new byte[BYTES_TO_READ];
+
+                Marshal.Copy(pointer, currentInput, 0, BYTES_TO_READ);
+            **/
+            /** UNSAFE CODE */
+            if (pointer == IntPtr.Zero)
+            {
+                return;
+            }
+            Span<byte> fromDeviceData = new Span<byte>((void*)pointer, BYTES_TO_READ);
+            Span<byte> currentInput = stackalloc byte[BYTES_TO_READ];
+            fromDeviceData.CopyTo(currentInput);
+            ReadDataWithDebounce(currentInput);
+            /** UNSAFE CODE */
+        }
+        public override void ReadDataWithDebounce(ReadOnlySpan<byte> data)
+        {
+            // Check if the state has changed
+            //if (ByteArraysEqual(_currentState, data))

```

Comment:
why remove this check? if current state and data are equal there is no reason to process the packet? Removing this should make things less efficient.